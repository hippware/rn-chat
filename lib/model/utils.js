"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const mobx_1 = require("mobx");
function waitFor(condition) {
    return tslib_1.__awaiter(this, void 0, void 0, function* () {
        return new Promise((resolve, reject) => {
            mobx_1.when(() => {
                let res = false;
                try {
                    res = condition();
                }
                catch (e) {
                    reject(e);
                }
                return res;
            }, () => {
                resolve();
            });
        });
    });
}
exports.waitFor = waitFor;
//# sourceMappingURL=utils.js.map