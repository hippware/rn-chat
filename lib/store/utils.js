"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const mobx_1 = require("mobx");
function pad(n, width, z = '0') {
    const str = n.toString();
    return str.length >= width ? str : new Array(width - str.length + 1).join(z) + n;
}
function process(result) {
    if (typeof result === 'object') {
        if (Array.isArray(result)) {
            return result.map(el => process(el));
        }
        else {
            if (result['#text'] && Object.keys(result).length === 1) {
                return result['#text'];
            }
            const res = {};
            let changed = false;
            // eslint-disable-next-line
            for (const key in result) {
                // eslint-disable-next-line
                if (result.hasOwnProperty(key)) {
                    changed = true;
                    res[key] = process(result[key]);
                }
            }
            if (!changed) {
                return undefined;
            }
            return res;
        }
    }
    else {
        return result;
    }
}
function isArray(res) {
    return Array.isArray(res) || mobx_1.isObservableArray(res);
}
exports.isArray = isArray;
function camelize(str) {
    return str
        .replace(/\W|_|\d/g, ' ')
        .replace(/(?:^\w|[A-Z]|\b\w)/g, (letter, index) => {
        return index === 0 ? letter.toLowerCase() : letter.toUpperCase();
    })
        .replace(/\s+/g, '');
}
exports.camelize = camelize;
function processMap(data) {
    const res = {};
    Object.keys(data).forEach(key => {
        const value = data[key];
        try {
            if (value && value !== 'null' && key !== 'field') {
                if (key === 'roles') {
                    res.roles = isArray(data.roles.role) ? data.roles.role : [data.roles.role];
                }
                else if (['followers', 'bots', 'followed'].indexOf(key) !== -1) {
                    res[key + 'Size'] = parseInt(data[key].size);
                }
                else if (data[key].thumbnail_url !== undefined) {
                    // we have image here!
                    if (data[key]['#text']) {
                        res[key] = { id: data[key]['#text'], url: data[key].thumbnail_url };
                    }
                }
                else if (key === 'subscribed') {
                    res.isSubscribed = value === 'true';
                }
                else if (key === 'owner') {
                    res.owner = Strophe.getNodeFromJid(value);
                }
                else if (key === 'subscribers') {
                    res.followersSize = parseInt(value.size);
                }
                else if (key === 'location') {
                    res.location = { latitude: parseFloat(value.geoloc.lat), longitude: parseFloat(value.geoloc.lon) };
                }
                else if (key === 'updated') {
                    res.time = iso8601toDate(value).getTime();
                }
                else if (key === 'radius') {
                    res.radius = parseFloat(value);
                }
                else {
                    const numbers = ['image_items', 'total_items', 'visibility'];
                    res[camelize(key)] = numbers.indexOf(key) !== -1 ? parseInt(value) : value;
                }
            }
        }
        catch (e) {
            console.error(`Cannot process key ${key} value: ${value}`);
        }
    });
    delete res.id;
    return res;
}
exports.processMap = processMap;
function fromCamelCase(data = {}) {
    const { firstName, userID, phoneNumber, lastName, sessionID, uuid } = data, result = tslib_1.__rest(data, ["firstName", "userID", "phoneNumber", "lastName", "sessionID", "uuid"]);
    if (phoneNumber) {
        result.phone_number = phoneNumber;
        result.phoneNumber = phoneNumber;
    }
    if (userID) {
        result.auth_user = userID;
    }
    if (firstName) {
        result.first_name = firstName;
    }
    if (lastName) {
        result.last_name = lastName;
    }
    if (sessionID) {
        result.token = sessionID;
    }
    if (uuid) {
        result.user = uuid;
    }
    return result;
}
exports.fromCamelCase = fromCamelCase;
function getJid(username, host, resource) {
    let jid = `${username}@${host}`;
    if (resource) {
        jid = `${jid}/${resource}`;
    }
    return jid;
}
exports.getJid = getJid;
function getNodeJid(jid) {
    if (jid.indexOf('@') < 0) {
        return jid;
    }
    return jid.split('@')[0];
}
exports.getNodeJid = getNodeJid;
/** Function: getUniqueId
 *  Generate a unique ID for use in <iq/> elements.
 *
 *  All <iq/> stanzas are required to have unique id attributes.  This
 *  function makes creating these easy.  Each connection instance has
 *  a counter which starts from zero, and the value of this counter
 *  plus a colon followed by the suffix becomes the unique id. If no
 *  suffix is supplied, the counter is used as the unique id.
 *
 *  Suffixes are used to make debugging easier when reading the stream
 *  data, and their use is recommended.  The counter resets to 0 for
 *  every new connection for the same reason.  For connections to the
 *  same server that authenticate the same way, all the ids should be
 *  the same, which makes it easy to see changes.  This is useful for
 *  automated testing as well.
 *
 *  Parameters:
 *    (String) suffix - A optional suffix to append to the id.
 *
 *  Returns:
 *    A unique string to be used for the id attribute.
 */
function getUniqueId(suffix) {
    let uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, c => {
        let r = (Math.random() * 16) | 0, // eslint-disable-line
        v = c === 'x' ? r : (r & 0x3) | 0x8; // eslint-disable-line
        return v.toString(16);
    });
    if (typeof suffix === 'string' || typeof suffix === 'number') {
        return `${uuid}:${suffix}`;
    }
    else {
        return `${uuid}`;
    }
}
exports.getUniqueId = getUniqueId;
function generateID() {
    const time = Date.now();
    return `s${time}${pad(Math.round(Math.random() * 1000), 4)}`;
}
exports.generateID = generateID;
function parseXml(xml, arrayTags) {
    function isArray(o) {
        return Object.prototype.toString.apply(o) === '[object Array]';
    }
    function parseNode(xmlNode, res) {
        if (!xmlNode) {
            return;
        }
        if (xmlNode.nodeName === '#text') {
            /* if you want the object to have a properyty "#text" even if it is "",
                               remove that if-else and use code that is currently in else block
                               */
            if (xmlNode.nodeValue.trim() === '') {
                return;
            }
            else {
                res[xmlNode.nodeName] = xmlNode.nodeValue;
                return;
            }
        }
        const jsonNode = {};
        const existing = res[xmlNode.nodeName];
        if (existing) {
            if (!isArray(existing)) {
                res[xmlNode.nodeName] = [existing, jsonNode];
            }
            else {
                res[xmlNode.nodeName].push(jsonNode);
            }
        }
        else if (arrayTags && arrayTags.indexOf(xmlNode.nodeName) !== -1) {
            res[xmlNode.nodeName] = [jsonNode];
        }
        else {
            res[xmlNode.nodeName] = jsonNode;
        }
        if (xmlNode.attributes) {
            const len = xmlNode.attributes.length;
            for (let i = 0; i < len; i++) {
                const attribute = xmlNode.attributes[i];
                jsonNode[attribute.nodeName] = attribute.nodeValue;
            }
        }
        const length = xmlNode.childNodes.length;
        for (let i = 0; i < length; i++) {
            parseNode(xmlNode.childNodes[i], jsonNode);
        }
    }
    const result = {};
    parseNode(xml, result);
    return process(result);
}
exports.parseXml = parseXml;
// return hashcode for given string
function hashCode(s) {
    let hash = 0, i, chr, len;
    if (s.length === 0)
        return hash;
    for (i = 0, len = s.length; i < len; i++) {
        chr = s.charCodeAt(i);
        hash = (hash << 5) - hash + chr; // eslint-disable-line
        // Convert to 32bit integer.
        hash = Math.abs(hash & hash); // eslint-disable-line
    }
    return hash;
}
exports.hashCode = hashCode;
function iso8601toDate(date) {
    let timestamp = Date.parse(date), minutesOffset = 0;
    if (isNaN(timestamp)) {
        // eslint-disable-next-line
        let struct = /^(\d{4}|[+\-]\d{6})-(\d{2})-(\d{2})(?:[T ](\d{2}):(\d{2})(?::(\d{2})(?:\.(\d{3,}))?)?(?:(Z)|([+\-])(\d{2})(?::?(\d{2}))?))?/.exec(date);
        if (struct) {
            if (struct[8] !== 'Z') {
                minutesOffset = +struct[10] * 60 + +struct[11];
                if (struct[9] === '+') {
                    minutesOffset = -minutesOffset;
                }
            }
            return new Date(+struct[1], +struct[2] - 1, +struct[3], +struct[4], +struct[5] + minutesOffset, +struct[6], struct[7] ? +struct[7].substr(0, 3) : 0);
        }
        else {
            // XEP-0091 dateAsString
            timestamp = Date.parse(`${date.replace(/^(\d{4})(\d{2})(\d{2})/, '$1-$2-$3')}Z`);
        }
    }
    return new Date(timestamp);
}
exports.iso8601toDate = iso8601toDate;
//# sourceMappingURL=utils.js.map