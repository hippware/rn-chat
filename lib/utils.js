"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
function process(result) {
    if (typeof result === 'object') {
        if (Array.isArray(result)) {
            return result.map(function (el) { return process(el); });
        }
        else {
            if (result['#text'] && Object.keys(result).length === 1) {
                return result['#text'];
            }
            var res = {};
            var changed = false;
            // eslint-disable-next-line
            for (var key in result) {
                // eslint-disable-next-line
                if (result.hasOwnProperty(key)) {
                    changed = true;
                    res[key] = process(result[key]);
                }
            }
            if (!changed) {
                return undefined;
            }
            return res;
        }
    }
    else {
        return result;
    }
}
exports.default = {
    fromCamelCase: function (data) {
        if (data === void 0) { data = {}; }
        var firstName = data.firstName, userID = data.userID, phoneNumber = data.phoneNumber, lastName = data.lastName, sessionID = data.sessionID, uuid = data.uuid, result = tslib_1.__rest(data, ["firstName", "userID", "phoneNumber", "lastName", "sessionID", "uuid"]);
        if (phoneNumber) {
            result.phone_number = phoneNumber;
            result.phoneNumber = phoneNumber;
        }
        if (userID) {
            result.auth_user = userID;
        }
        if (firstName) {
            result.first_name = firstName;
        }
        if (lastName) {
            result.last_name = lastName;
        }
        if (sessionID) {
            result.token = sessionID;
        }
        if (uuid) {
            result.user = uuid;
        }
        return result;
    },
    getJid: function (username, host, resource) {
        var jid = username + "@" + host;
        if (resource) {
            jid = jid + "/" + resource;
        }
        return jid;
    },
    getNodeJid: function (jid) {
        if (jid.indexOf('@') < 0) {
            return null;
        }
        return jid.split('@')[0];
    },
    /** Function: getUniqueId
     *  Generate a unique ID for use in <iq/> elements.
     *
     *  All <iq/> stanzas are required to have unique id attributes.  This
     *  function makes creating these easy.  Each connection instance has
     *  a counter which starts from zero, and the value of this counter
     *  plus a colon followed by the suffix becomes the unique id. If no
     *  suffix is supplied, the counter is used as the unique id.
     *
     *  Suffixes are used to make debugging easier when reading the stream
     *  data, and their use is recommended.  The counter resets to 0 for
     *  every new connection for the same reason.  For connections to the
     *  same server that authenticate the same way, all the ids should be
     *  the same, which makes it easy to see changes.  This is useful for
     *  automated testing as well.
     *
     *  Parameters:
     *    (String) suffix - A optional suffix to append to the id.
     *
     *  Returns:
     *    A unique string to be used for the id attribute.
     */
    getUniqueId: function (suffix) {
        var uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
            var r = (Math.random() * 16) | 0, // eslint-disable-line
            v = c === 'x' ? r : (r & 0x3) | 0x8; // eslint-disable-line
            return v.toString(16);
        });
        if (typeof suffix === 'string' || typeof suffix === 'number') {
            return uuid + ":" + suffix;
        }
        else {
            return "" + uuid;
        }
    },
    parseXml: function (xml, arrayTags) {
        function isArray(o) {
            return Object.prototype.toString.apply(o) === '[object Array]';
        }
        function parseNode(xmlNode, res) {
            if (!xmlNode) {
                return;
            }
            if (xmlNode.nodeName === '#text') {
                /* if you want the object to have a properyty "#text" even if it is "",
                                 remove that if-else and use code that is currently in else block
                                 */
                if (xmlNode.nodeValue.trim() === '') {
                    return;
                }
                else {
                    res[xmlNode.nodeName] = xmlNode.nodeValue;
                    return;
                }
            }
            var jsonNode = {};
            var existing = res[xmlNode.nodeName];
            if (existing) {
                if (!isArray(existing)) {
                    res[xmlNode.nodeName] = [existing, jsonNode];
                }
                else {
                    res[xmlNode.nodeName].push(jsonNode);
                }
            }
            else if (arrayTags && arrayTags.indexOf(xmlNode.nodeName) !== -1) {
                res[xmlNode.nodeName] = [jsonNode];
            }
            else {
                res[xmlNode.nodeName] = jsonNode;
            }
            if (xmlNode.attributes) {
                var len = xmlNode.attributes.length;
                for (var i = 0; i < len; i++) {
                    var attribute = xmlNode.attributes[i];
                    jsonNode[attribute.nodeName] = attribute.nodeValue;
                }
            }
            var length = xmlNode.childNodes.length;
            for (var i = 0; i < length; i++) {
                parseNode(xmlNode.childNodes[i], jsonNode);
            }
        }
        var result = {};
        parseNode(xml, result);
        return process(result);
    },
    // return hashcode for given string
    hashCode: function (s) {
        var hash = 0, i, chr, len;
        if (s.length === 0)
            return hash;
        for (i = 0, len = s.length; i < len; i++) {
            chr = s.charCodeAt(i);
            hash = (hash << 5) - hash + chr; // eslint-disable-line
            // Convert to 32bit integer.
            hash = Math.abs(hash & hash); // eslint-disable-line
        }
        return hash;
    },
    iso8601toDate: function (date) {
        var timestamp = Date.parse(date), minutesOffset = 0;
        if (isNaN(timestamp)) {
            // eslint-disable-next-line
            var struct = /^(\d{4}|[+\-]\d{6})-(\d{2})-(\d{2})(?:[T ](\d{2}):(\d{2})(?::(\d{2})(?:\.(\d{3,}))?)?(?:(Z)|([+\-])(\d{2})(?::?(\d{2}))?))?/.exec(date);
            if (struct) {
                if (struct[8] !== 'Z') {
                    minutesOffset = +struct[10] * 60 + +struct[11];
                    if (struct[9] === '+') {
                        minutesOffset = -minutesOffset;
                    }
                }
                return new Date(+struct[1], +struct[2] - 1, +struct[3], +struct[4], +struct[5] + minutesOffset, +struct[6], struct[7] ? +struct[7].substr(0, 3) : 0);
            }
            else {
                // XEP-0091 dateAsString
                timestamp = Date.parse(date.replace(/^(\d{4})(\d{2})(\d{2})/, '$1-$2-$3') + "Z");
            }
        }
        return new Date(timestamp);
    }
};
//# sourceMappingURL=utils.js.map