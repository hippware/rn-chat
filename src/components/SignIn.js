// @flow

import React from 'react';
import {View, Image, StyleSheet, NativeModules, TouchableOpacity, Text, Keyboard} from 'react-native';
import {observer} from 'mobx-react/native';
import {observable} from 'mobx';
import {RText} from './common';
import {colors} from '../constants';
import {k} from './Global';
import CustomTextInput from './SignUpTextInput';
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
import CountryPicker, {getAllCountries} from 'react-native-country-picker-modal';
import Button from 'apsl-react-native-button';
import {Actions} from 'react-native-router-flux';
import firebaseStore from '../store/firebaseStore';

// const phoneUtil = require('google-libphonenumber').PhoneNumberUtil.getInstance();
import {parse, format, asYouType} from 'libphonenumber-js';

// const parsed = phoneUtil.parse('202-456-1414', 'US');
// const isValid = phoneUtil.isValidNumber(parsed);
// console.log('& phone util', isValid, parsed);
const CarrierInfo = NativeModules.RNCarrierInfo;

const countryMap = {};
getAllCountries().forEach(country => (countryMap[country.cca2] = country));

type Props = {
  error?: string,
  onVerify: Function,
};

@observer
class SignIn extends React.Component {
  props: Props;
  picker: any;
  @observable cca2: string = 'US';
  @observable callingCode: string = '1';
  @observable countryName: string = 'United States';
  @observable phoneValue: string = '';
  @observable submitting: boolean = false;
  @observable sendText: string = 'Send Confirmation';
  phoneText: any;

  componentDidMount() {
    CarrierInfo.isoCountryCode((result) => {
      if (result && result !== 'nil') {
        this.cca2 = result.toUpperCase();
        const data = countryMap[this.cca2];
        this.callingCode = data.callingCode;
        this.countryName = data.name.common;
      }
    });
  }

  processText = (text: string) => {
    const parsed = parse(text, this.cca2);
    this.phoneValue = /\d{4,}/.test(text) ? new asYouType(this.cca2).input(text) : text;
    if (parsed.country && parsed.phone) {
      this.phoneText.valid = true;
    } else {
      this.phoneText.valid = false;
    }
  };

  submit = async () => {
    if (!this.phoneText.valid) {
      this.phoneText.message = 'Please check your phone number and try again';
    } else {
      this.submitting = true;
      this.phoneText.message = '';
      Keyboard.dismiss();
      // await this.props.onVerify({phone: `+${this.callingCode}${this.phoneValue.replace(/\D/g, '')}`});
      try {
        await firebaseStore.verifyPhone({phone: `+${this.callingCode}${this.phoneValue.replace(/\D/g, '')}`});
        Actions.verifyCode();
      } catch (err) {
        console.warn('verify phone error', err);
        let message;
        switch (err.code) {
          case 'auth/too-many-requests':
            message = 'Too many login attempts from this phone. Try again later.';
            break;
          case 'auth/network-request-failed':
            message = 'Network error. Check your connection and try again.';
            break;
          default:
            message = err.message;
        }
        alert(message);
      } finally {
        this.submitting = false;
      }
    }
  };

  render() {
    return (
      <KeyboardAwareScrollView style={{flex: 1, backgroundColor: colors.WHITE}} keyboardShouldPersistTaps='always'>
        <View style={{flexDirection: 'row', marginLeft: 60 * k, marginTop: 32 * k}}>
          <Image style={[{width: 60, height: 69}]} resizeMode='contain' source={require('../../images/iconBot.png')} />
          <View style={{marginLeft: 20, marginTop: -5}}>
            <RText size={28} weight='Light' color={colors.PINK}>
              {'Please verify\r\nyour phone\r\nnumber.'}
            </RText>
            <RText size={15} color={colors.DARK_GREY} style={{marginTop: 7 * k}}>
              {"Don't worry we won't share\r\nyour phone number."}
            </RText>
            {this.props.error &&
              <RText size={15} color={'red'} style={{marginTop: 7 * k, paddingRight: 120 * k}}>
                {this.props.error}
              </RText>}
          </View>
        </View>
        <View style={{marginTop: 20 * k}}>
          <CountryPicker
            // countryList={NORTH_AMERICA}
            onChange={(value) => {
              this.cca2 = value.cca2;
              this.callingCode = value.callingCode;
              this.countryName = value.name;
            }}
            cca2={this.cca2}
            translation='eng'
            filterable
            closeable
            ref={r => (this.picker = r)}
          >
            <TouchableOpacity onPress={() => this.picker.openModal()}>
              <CustomTextInput
                icon={require('../../images/globe.png')}
                label='Country Code'
                autoCapitalize='none'
                validate={() => {}}
                value={`${this.countryName} +${this.callingCode}`}
                editable={false}
                pointerEvents='none'
              />
            </TouchableOpacity>
          </CountryPicker>

          <CustomTextInput
            icon={require('../../images/phone.png')}
            label='Phone Number'
            autoFocus
            autoCorrect={false}
            keyboardType='phone-pad'
            ref={r => (this.phoneText = r)}
            onChangeText={this.processText}
            value={this.phoneValue}
          />

          <View style={{marginHorizontal: 36 * k, marginVertical: 20 * k}}>
            <Button style={styles.button} isDisabled={this.submitting || !this.phoneText || !this.phoneText.valid} onPress={this.submit} textStyle={styles.text}>
              {this.submitting ? 'Sending...' : 'Send Confirmation'}
            </Button>
            <RText size={12.5} color={colors.DARK_GREY} style={{marginTop: 10 * k, textAlign: 'center'}}>
              {'By tapping "Send Confirmation", we will send you a SMS. Message, data rates may apply.'}
            </RText>
          </View>
        </View>
      </KeyboardAwareScrollView>
    );
  }
}

export default SignIn;

const styles = StyleSheet.create({
  button: {
    flex: 1,
    height: 50 * k,
    borderWidth: 0,
    borderRadius: 5 * k,
    backgroundColor: colors.PINK,
    alignItems: 'center',
    marginHorizontal: 5 * k,
    justifyContent: 'center',
  },
  text: {fontSize: 17.5 * k, letterSpacing: 0.8, fontFamily: 'Roboto-Regular', color: colors.WHITE},
});
