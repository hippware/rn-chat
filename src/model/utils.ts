import {when} from 'mobx'

export async function waitFor(condition: () => boolean) {
  return new Promise((resolve, reject) => {
    when(
      () => {
        let res = false
        try {
          res = condition()
        } catch (e) {
          reject(e)
        }
        return res
      },
      () => {
        resolve()
      }
    )
  })
}
