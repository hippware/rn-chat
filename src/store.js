import { compose, createStore, applyMiddleware } from 'redux';
import createLogger from 'redux-logger'
import reducer from './reducers/root';
import { persistStore, autoRehydrate } from 'redux-persist'
const loggerMiddleware = createLogger();
import {PERSIST, DEBUG} from './globals';
const createStoreWithMiddleware = DEBUG ? applyMiddleware(loggerMiddleware)(createStore) : createStore;
import { createEffectCapableStore } from 'redux-side-effects';

export default function(persist){
  const createPersistStore = persist ? compose(autoRehydrate())(createStoreWithMiddleware) : createStoreWithMiddleware;
  const createEffectStoreWithMiddleware = createEffectCapableStore(createPersistStore);
  return createEffectStoreWithMiddleware(reducer);

}
