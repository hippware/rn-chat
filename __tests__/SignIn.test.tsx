import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import SignIn from '../src/components/SignIn'
import './utils/mockTextInput'
import {IFirebaseStore} from 'src/store/FirebaseStore'

describe('SignIn', () => {
  it('renders', () => {
    const firebaseStore = {} as IFirebaseStore
    const tree = renderer.create(<SignIn firebaseStore={firebaseStore} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
