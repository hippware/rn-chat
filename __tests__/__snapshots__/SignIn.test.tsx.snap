// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="always"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "rgb(255,255,255)",
      "flex": 1,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginLeft": 120,
          "marginTop": 64,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../images/iconBot.png",
          }
        }
        style={
          Array [
            Object {
              "height": 69,
              "width": 60,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "marginLeft": 20,
            "marginTop": -5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(253,95,108)",
                "fontFamily": "Roboto-Light",
                "fontSize": 28,
              },
              undefined,
            ]
          }
        >
          Please verify
your phone
number.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(155,155,155)",
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
              },
              Object {
                "marginTop": 14,
              },
            ]
          }
        >
          Don't worry we won't share
your phone number.
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 40,
        }
      }
    >
      <CountryPicker
        cca2="US"
        closeable={true}
        filterable={true}
        onChange={[Function]}
        translation="eng"
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "height": 10,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingBottom": 13,
                    "paddingHorizontal": 10,
                    "paddingTop": 9,
                  },
                  Object {
                    "justifyContent": "center",
                    "paddingTop": 2,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../images/globe.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "marginLeft": 16,
                        "marginRight": 13,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <TextInput
                  autoCapitalize="none"
                  autoCorrect={false}
                  clearButtonMode="while-editing"
                  editable={false}
                  icon={
                    Object {
                      "testUri": "../../../images/globe.png",
                    }
                  }
                  label="Country Code"
                  onChangeText={[Function]}
                  placeholder="Country Code"
                  placeholderTextColor="rgb(212,212,212)"
                  pointerEvents="none"
                  returnKeyType="next"
                  style={
                    Object {
                      "color": "rgb(63,50,77)",
                      "flex": 1,
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 18,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value="United States +1"
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "width": 30,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(63, 50, 77, .2)",
                    "height": 0.5,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </CountryPicker>
      <View>
        <View
          style={
            Object {
              "height": 10,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingBottom": 13,
                "paddingHorizontal": 10,
                "paddingTop": 9,
              },
              Object {
                "justifyContent": "center",
                "paddingTop": 2,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../images/phone.png",
                }
              }
              style={
                Array [
                  Object {
                    "marginLeft": 16,
                    "marginRight": 13,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <TextInput
              autoCorrect={false}
              autoFocus={true}
              clearButtonMode="while-editing"
              icon={
                Object {
                  "testUri": "../../../images/phone.png",
                }
              }
              keyboardType="phone-pad"
              label="Phone Number"
              onChangeText={[Function]}
              placeholder="Phone Number"
              placeholderTextColor="rgb(212,212,212)"
              returnKeyType="next"
              style={
                Object {
                  "color": "rgb(63,50,77)",
                  "flex": 1,
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 18,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": 30,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(63, 50, 77, .2)",
                "height": 0.5,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 72,
            "marginVertical": 40,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(253,95,108)",
              "borderRadius": 10,
              "borderWidth": 0,
              "flex": 1,
              "height": 100,
              "justifyContent": "center",
              "marginHorizontal": 10,
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 12,
                },
                Object {
                  "color": "rgb(255,255,255)",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 35,
                  "letterSpacing": 0.8,
                },
              ]
            }
          >
            Send Confirmation
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(155,155,155)",
                "fontFamily": "Roboto-Regular",
                "fontSize": 12.5,
              },
              Object {
                "marginTop": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          By tapping "Send Confirmation", we will send you a SMS. Message, data rates may apply.
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
